vaadin-menu-bar-flow-parent/vaadin-menu-bar-testbench/src/main/java/com/vaadin/flow/component/menubar/testbench/MenuBarElement.java

```diff
@@ -84,41 +83,38 @@ public List<MenuBarItemElement> getSubMenuItems() {
      * Get TestBenchElements representing sub menu items under specific sub
      * menu.
      *
-     * @param overlay
-     *            The sub menu overlay from which items are being collected.
+     * @param subMenu
+     *            The sub menu from which items are being collected.
      * @return List of MenuBarItemElement representing sub menu items.
      */
-    public List<MenuBarItemElement> getSubMenuItems(TestBenchElement overlay) {
-        return overlay.$(MenuBarItemElement.class).all();
+    public List<MenuBarItemElement> getSubMenuItems(TestBenchElement subMenu) {
+        return subMenu.getPropertyElement("_listBox")
+                .$(MenuBarItemElement.class).all();
     }
 
     /**
-     * Get the sub menu overlay element.
+     * Get the sub menu element.
      *
-     * @return TestBenchElement for the first open sub menu in this menu bar
+     * @return TestBenchElement for the first sub menu in this menu bar
      */
     public TestBenchElement getSubMenu() {
-        var button = $(MenuBarButtonElement.class).withAttribute("expanded")
-                .withCondition(this::isVisible).first();
-        return button != null ? button.getSubMenu() : null;
+        return getPropertyElement("_subMenu");
     }
 
     /**
-     * Get all the open sub menu overlay elements.
+     * Get all the opened sub menu elements.
      *
-     * @return List of TestBenchElements representing currently open sub menus.
+     * @return List of TestBenchElements representing all opened sub menus.
      */
     public List<TestBenchElement> getAllSubMenus() {
-        waitForSubMenu();
         List<TestBenchElement> elements = new ArrayList<>();
-        getDriver().findElements(By.tagName(OVERLAY_TAG))
-                .forEach(element -> elements.add((TestBenchElement) element));
+        findElements(By.tagName(SUBMENU_TAG)).forEach(element -> {
+            if (element.getDomProperty("opened")
+                    .equals(Boolean.TRUE.toString())) {
+                elements.add((TestBenchElement) element);
+            }
+        });
         return elements;
```

Comment:
```suggestion
        return $(SUBMENU_TAG).withAttribute("opened").all();
```
That should do the same?